egs/tedlium/s5_r2/local/rnnlm/tuning/run_lstm_tdnn_with_lm1b.sh

```diff
@@ -0,0 +1,156 @@
+lstm_rpd=256
+lstm_nrpd=256
+stage=0
+train_stage=-10
+epochs=20
+
+# variables for lattice rescoring
+run_lat_rescore=true
+run_nbest_rescore=true
+decode_dir_suffix=rnnlm_lm1b_tedlium_weight3
+ac_model_dir=/export/a12/ywang/kaldi/egs/tedlium/s5_r2/exp/chain_cleaned/tdnn_lstm1i_adversarial1.0_interval4_epoches7_lin_to_5_sp_bi
+ngram_order=4 # approximate the lattice-rescoring by limiting the max-ngram-order
+              # if it's set, it merges histories in the lattice if they share
+              # the same ngram history and this prevents the lattice from 
+              # exploding exponentially
+pruned_rescore=true
+
+. ./cmd.sh
+. ./utils/parse_options.sh
+
+text=/export/a07/keli1/1-billion-word-language-modeling-benchmark/training-monolingual.tokenized.shuffled/news.en
+wordlist=data/lang/words.txt
+train_text=data/train/text
+dev_sents=10000
+text_dir=data/rnnlm/text_lm1b_tedlium
+mkdir -p $dir/config
+set -e
+
+for f in $text $wordlist $train_text; do
+  [ ! -f $f ] && \
+    echo "$0: expected file $f to exist; generate lm1b data first; \
+    search for local/prepare_data.sh and utils/prepare_lang.sh in run.sh" && exit 1
+done
+
+if [ $stage -le 0 ]; then
+  mkdir -p $text_dir
+  cat $train_text | cut -d ' ' -f2- | head -n $dev_sents > $text_dir/dev.txt
+  cat $train_text | cut -d ' ' -f2- | tail -n +$[$dev_sents+1] > $text_dir/ted.txt
+  cp $text $text_dir/lm1b.txt
+fi
+
+
+if [ $stage -le 1 ]; then
+  cp $wordlist $dir/config/
+  n=`cat $dir/config/words.txt | wc -l`
+  echo "<brk> $n" >> $dir/config/words.txt
+
+  # words that are not present in words.txt but are in the training or dev data, will be
+  # mapped to <unk> during training.
+  echo "<unk>" >$dir/config/oov.txt
+
+  cat > $dir/config/data_weights.txt <<EOF
+ted   1   3.0
+lm1b    1   1.0
+EOF
+
+  rnnlm/get_unigram_probs.py --vocab-file=$dir/config/words.txt \
+                             --unk-word="<unk>" \
+                             --data-weights-file=$dir/config/data_weights.txt \
+                             $text_dir | awk 'NF==2' >$dir/config/unigram_probs.txt
+
+  # choose features
+  rnnlm/choose_features.py --unigram-probs=$dir/config/unigram_probs.txt \
+                           --use-constant-feature=true \
+                           --top-word-features=10000 \
+                           --min-frequency 1.0e-03 \
+                           --special-words='<s>,</s>,<brk>,<unk>' \
+                           $dir/config/words.txt > $dir/config/features.txt
+fi
+
+cat >$dir/config/xconfig <<EOF
+input dim=$embedding_dim name=input
+relu-renorm-layer name=tdnn1 dim=$embedding_dim input=Append(0, IfDefined(-1))
+fast-lstmp-layer name=lstm1 cell-dim=$embedding_dim recurrent-projection-dim=$lstm_rpd non-recurrent-projection-dim=$lstm_nrpd
+relu-renorm-layer name=tdnn2 dim=$embedding_dim input=Append(0, IfDefined(-2))
+fast-lstmp-layer name=lstm2 cell-dim=$embedding_dim recurrent-projection-dim=$lstm_rpd non-recurrent-projection-dim=$lstm_nrpd
```

Comment:
I'll merge this but I think you should play with the l2-regularize option for both the fast-lstmp-layer and the relu-renorm-layers.  See my email on the list to that guy who had problems with long utterances, for tuning suggestions.   Right now there is no shrinkage or anything that would keep the LSTM parameters in the right range.